File: src/define/process.go
package define

//ProcessMonitor 进程监控接口字段
type ProcessMonitor struct {
	Exe_file  string `json:"exe_file"`	//进程文件
	Exe_hash  string `json:"exe_hash"`	//文件hash
	Pid       uint32 `json:"pid"`		//进程id
	Ppid      uint32 `json:"ppid"`		//父进程ID
	Data_type uint32 `json:"data_type" enum:"1001,1002,1003"`	//数据类型
	Argv      string `json:"argv"`		//进程参数
	Uid       uint32 `json:"uid"`		//用户ID
	Gid       uint32 `json:"gid"`		//用户组ID
	Namespace uint32 `json:"namespace"`	//进程命名空间ID
	Timestamp uint64 `json:"timestamp"`	//事件时间戳
}

File: akfs/process.c
#include "process.h"
#include "togo.h"

#define PsUsedPort  65431

/**
 * @brief process_callback
 *   回调处理模块
 */
static int process_callback(unsigned char *buffer ,unsigned int size)
{
    send2Go(buffer,size,PsUsedPort);
//
//    akfs_process_t *p = NULL;
//
//    p = (akfs_process_t *)buffer;
//    printf("type:[%d] pid %d ppid %d tgid %d tpath:[%s] ns:[%u] args:[%s] hash:[%s] timestamp[%s]\n" ,p->data_type ,p->pid ,p->ppid ,p->gid ,
//        p->exec_file ,p->ns ,p->argv ,p->exec_hash ,p->timestamp);

    return 0;
}

int ProcessMonitor(void)
{
    int ret = 0;
    akfs_t gat;

    ret = akfs_open(&gat ,"/opt/mount/process");
    if(ret){return ret;}

    ret = akfs_get_access(&gat);
    if(ret){
        akfs_close(&gat);
        goto out;
    }

    akfs_loop_read(&gat ,process_callback);

    akfs_close(&gat);
out:
    return 0;
}